benchmarks/linear_programming/run_mps_files.sh

```diff
@@ -0,0 +1,276 @@
+
+
+# This script runs MPS (Mathematical Programming System) files in parallel across multiple GPUs
+# It supports running linear programming (LP) and mixed-integer programming (MIP) problems
+#
+# Usage:
+#   ./run_mps_files.sh --path /path/to/mps/files --ngpus 4 --time-limit 3600 --output-dir results
+#
+# Arguments:
+#   --path        : Directory containing .mps files (required)
+#   --ngpus       : Number of GPUs to use (default: 1)
+#   --time-limit  : Time limit in seconds for each problem (default: 360)
+#   --output-dir  : Directory to store output log files (default: current directory)
+#   --relaxation  : Run relaxation instead of solving the MIP
+#   --mip-heuristics-only : Run mip heuristics only
+#   --write-log-file : Write log file
+#   --num-cpu-threads : Number of CPU threads to use
+#   --batch-num : Batch number
+#   --n-batches : Number of batches
+#   --log-to-console : Log to console
+#
+# Examples:
+#   # Run all MPS files in /data/lp using 2 GPUs with 1 hour time limit
+#   ./run_mps_files.sh --path /data/lp --ngpus 2 --time-limit 3600
+#
+#   # Run with specific GPU devices using CUDA_VISIBLE_DEVICES
+#   CUDA_VISIBLE_DEVICES=0,2,3 ./run_mps_files.sh --path /data/mip
+#
+#   # Run with custom output directory
+#   ./run_mps_files.sh --path /data/lp --output-dir ~/results/lp_benchmark
+#
+# Notes:
+#   - Files are distributed dynamically across available GPUs for load balancing
+#   - Each problem's results are logged to a separate file in the output directory
+#   - The script uses file locking to safely distribute work across parallel processes
+#   - If CUDA_VISIBLE_DEVICES is set, it can override the --ngpus argument
+
+# Help function
+print_help() {
+    cat << EOF
+This script runs MPS (Mathematical Programming System) files in parallel across multiple GPUs.
+It supports running linear programming (LP) and mixed-integer programming (MIP) problems.
+
+Usage:
+    $(basename "$0") --path /path/to/mps/files [options]
+
+Required Arguments:
+    --path PATH          Directory containing .mps files
+
+Optional Arguments:
+    --ngpus N           Number of GPUs to use (default: 1)
+    --time-limit N      Time limit in seconds for each problem (default: 360)
+    --output-dir PATH   Directory to store output log files (default: current directory)
+    --relaxation       Run relaxation instead of solving the MIP
+    --mip-heuristics-only  Run mip heuristics only
+    --write-log-file   Write log file
+    --num-cpu-threads  Number of CPU threads to use
+    --batch-num        Batch number
+    --n-batches        Number of batches
+    --log-to-console   Log to console
+    -h, --help         Show this help message and exit
+
+Examples:
+    # Run all MPS files in /data/lp using 2 GPUs with 1 hour time limit
+    $(basename "$0") --path /data/lp --ngpus 2 --time-limit 3600
+
+    # Run with specific GPU devices using CUDA_VISIBLE_DEVICES
+    CUDA_VISIBLE_DEVICES=0,2,3 $(basename "$0") --path /data/mip
```

Comment:
Since mps files don't have option to set solver settings, do you want to make this into config + dataset ? Does it makes sense to have a config to set solver settings for each  mps ?

Reply:
It definitely does not make sense to have different setting for each mps, because this script is purely for benchmarking purpose.  

With regards to settings, I am thinking this would be benchmarking with all the default settings.    That's why I did not add any parameter to it.

Reply:
Makes sense